agent/kubernetes/service.go

```diff
@@ -18,6 +18,24 @@ type CostService struct {
 	influxdbClient client.Client
 }
 
+// PodCostComponents struct to hold all the cost information for a single pod
+type PodCostComponents struct {
+	totalCost        decimal.Decimal
+	utilization      decimal.Decimal
+	underUtilization decimal.Decimal
+	nodeOverhead     decimal.Decimal
+}
+
+// NewPodCostComponents creates a new PodCostComponents
+func NewPodCostComponents(totalCost decimal.Decimal, utilization decimal.Decimal, underUtilization decimal.Decimal, nodeOverhead decimal.Decimal) *PodCostComponents {
```

Comment:
Shouldn't be a need for this, you can just create a `PodCostComponents` where you need it and assign to it as necessary.

Reply:
Rookie !